plugins/cache-caffeine/src/test/java/org/opensearch/cache/store/CaffeineHeapCacheTests.java

```diff
@@ -411,8 +376,73 @@ public void testConcurrentEvictions() throws Exception {
         phaser.arriveAndAwaitAdvance(); // Will trigger parallel puts above.
         countDownLatch.await(); // Wait for all threads to finish
 
-        assertEquals(randomKeys - 10, caffeineTest.stats().getTotalEvictions());
-        assertEquals(100, caffeineTest.stats().getTotalSizeInBytes());
+        ((CaffeineHeapCache<String, String>) caffeineTest).cleanUp(); // Manually performs maintenance cycle, which includes removing
+                                                                      // expired entries.
+        assertEquals(randomKeys - (TEST_MAX_CACHE_SIZE / MOCK_WEIGHT), caffeineTest.stats().getTotalEvictions());
+        assertEquals(randomKeys - (TEST_MAX_CACHE_SIZE / MOCK_WEIGHT), removalListener.evictionMetric.count());
+        assertEquals(TEST_MAX_CACHE_SIZE, caffeineTest.stats().getTotalSizeInBytes());
+    }
+
+    public void testReplace() throws Exception {
+        ToLongBiFunction<ICacheKey<String>, String> weigher = getMockWeigher(true);
+        MockRemovalListener<String, String> removalListener = new MockRemovalListener<>();
+        ICache<String, String> caffeineTest = new CaffeineHeapCache.Builder<String, String>().setDimensionNames(List.of(dimensionName))
+            .setExecutor(Runnable::run)
+            .setExpireAfterAccess(TimeValue.MAX_VALUE)
+            .setMaximumWeightInBytes(TEST_MAX_CACHE_SIZE)
+            .setWeigher(weigher)
+            .setRemovalListener(removalListener)
+            .build();
+
+        for (int i = 0; i < TEST_MAX_CACHE_SIZE / MOCK_WEIGHT; i++) {
+            ICacheKey<String> key = getICacheKey(UUID.randomUUID().toString());
+            String value = "10";
+            caffeineTest.put(key, value);
+        }
+        assertEquals(TEST_MAX_CACHE_SIZE, caffeineTest.stats().getTotalSizeInBytes());
+
+        // Putting a value of size "1" will evict one of the entries with value of size "10".
+        // Subsequent puts of size "1" values will not evict any more entries, since total size in bytes should be less than
+        // TEST_MAX_CACHE_SIZE.
+        int randomKeys = randomIntBetween(1, TEST_MAX_CACHE_SIZE / MOCK_WEIGHT);
+        for (int i = 0; i < randomKeys; i++) {
+            ICacheKey<String> key = getICacheKey(UUID.randomUUID().toString());
```

Comment:
I think this test doesn't actually test the `REPLACED` behavior, since that happens when we update the value for the same key. Here we pass in new values of `UUID.randomUUID().toString()`, so they'll be different every time, and we are actually testing the `EVICTED` behavior.